Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -944,6 +969,13 @@ public enum ListOrdering {
     case bottomToTop
 }
 
+// Describes the pin expiration
+public enum Pin {
+    case noExpiration
+    case expirationDate(Date)
+    case expirationTime(TimeInterval)
+}
+
```

Comment:
I think this simplifies a bit the API by adding this abstraction. Some customers might not know How to use the `Date` API. So with this abstraction, everyone will understand.

Also, we need to provide a way to `pin` and `unpin` a message. We can't just provide a date. Because if `nil` would mean `noExpiration` then how would you not pin a message, or unpin a message? This would require us to have a different property, like `pinned = true`. This is actually what the other platforms are doing but I think it is bad because it makes it possible to make a mistake and write `pinned = false` and `expirationDate = Date()`, which is a wrong state.

With this enum, we can describe 4 states. 
- No pin (`Pin? = nil`)
- Pin without expiration
- Pin with an expiration date
- Pin with the amount of time.